<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Cell Type Assignment • AtlasCortexSD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Cell Type Assignment">
<meta property="og:description" content="AtlasCortexSD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">AtlasCortexSD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_ct_anno.html">Cell Type Assignment</a>
    </li>
    <li>
      <a href="../articles/2_pb_dgea.html">Pseudo Bulk Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/3_bulk_de.html">Bulk RNAseq Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/4_Dtx_usage.html">Differential Transcripts Usage</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RissoLab/AtlasCortexSD" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Cell Type Assignment</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RissoLab/AtlasCortexSD/blob/HEAD/vignettes/1_ct_anno.Rmd" class="external-link"><code>vignettes/1_ct_anno.Rmd</code></a></small>
      <div class="hidden name"><code>1_ct_anno.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="preliminaries">Preliminaries<a class="anchor" aria-label="anchor" href="#preliminaries"></a>
</h2>
<div class="section level3">
<h3 id="data-availability">Data Availability<a class="anchor" aria-label="anchor" href="#data-availability"></a>
</h3>
<p>Raw sequencing data for this analysis is stored in GEO under
accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE211088" class="external-link">GSE211088</a>.</p>
<p>The code used in this analysis has been deposited into Github, and
can be available <a href="https://github.com/PeixotoLab/RNAseq_sleep/tree/main/02_analysis_snrnaseq" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="bioconductor-packages">Bioconductor packages<a class="anchor" aria-label="anchor" href="#bioconductor-packages"></a>
</h3>
<p>To identify mitochondrial genes, we retrieve the chromosome location
of each Ensembl gene with the <em><a href="https://bioconductor.org/packages/3.18/EnsDb.Mmusculus.v79" class="external-link">EnsDb.Mmusculus.v79</a></em>.
package. We used the Bioconductor <em><a href="https://bioconductor.org/packages/3.18/scuttle" class="external-link">scuttle</a></em>package
to detect low quality and damaged droplets. The potential doublets
removal was implemented in <em><a href="https://bioconductor.org/packages/3.18/scuttle" class="external-link">scuttle</a></em>
package. For cell-type assignment reference dataset was made available
at <a href="https://github.com/drighelli/AllenInstituteBrainData" class="external-link"><em>AllenInstituteBrainData</em></a>.
Cell annotation was computed using two methods: <a href="https://github.com/satijalab/azimuth" class="external-link"><em>Azimuth</em></a> and
<em><a href="https://bioconductor.org/packages/3.18/SingleR" class="external-link">SingleR</a></em>.
We used the <em><a href="https://bioconductor.org/packages/3.18/scuttle" class="external-link">scuttle</a></em>
package to create pseudo-bulk. The visualization of the MDS plot was
implemented in <em><a href="https://bioconductor.org/packages/3.18/muscat" class="external-link">muscat</a></em>
package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library(Seurat) library(SingleCellExperiment)</span></span>
<span><span class="co"># library(EnsDb.Mmusculus.v79) library(scuttle) library(scran)</span></span>
<span><span class="co"># library(scDblFinder) library(ggplot2) library(biomaRt)</span></span>
<span><span class="co"># library(muscat) library(edgeR) library(dplyr)</span></span>
<span><span class="co"># library(AtlasCortexSD) Dependencies necessary to reproduce the</span></span>
<span><span class="co"># entire code, not used in the website library(Azimuth)</span></span>
<span><span class="co"># library(SeuratData) library(AllenInstituteBrainData)</span></span></code></pre></div>
<!-- # Data pre-processing -->
<!-- ## Data setting -->
<!-- Load single-nuclear RNA-seq dataset.  -->
<!-- ```{r, eval=FALSE} -->
<!-- url <- "https://zenodo.org/records/10078998/files/sce_mouse_sleep_snrnaseq_complete.rds" -->
<!-- system(paste0("curl ", url)) -->
<!-- snrna_data <- readRDS("sce_mouse_sleep_snrnaseq_complete.rds") -->
<!-- snrna_data -->
<!-- ``` -->
<!-- We added the UMI counts of spliced mRNA and introns sharing the same Ensembl ID. To identify mitochondrial genes, we retrieve the chromosome location of each ensembl gene with the EnsDb.Mmusculus.v79 package. We split the data into six SingleCellExperiment objects, one for each mouse. -->
<!-- ```{r, eval=FALSE} -->
<!-- # Exons -->
<!-- exons <- snrna_data[-which(grepl("-I", rownames(snrna_data))), ] -->
<!-- rownames(exons) <- substring(rownames(exons), 1, 18) -->
<!-- introns <- snrna_data[which(grepl("-I", rownames(snrna_data))), ] -->
<!-- rownames(introns) <- substring(rownames(introns), 1, 18) -->
<!-- # We identified introns and exons with the same Ensembl ID -->
<!-- same_ensembl <- intersect(rownames(exons), rownames(introns)) -->
<!-- exons <- exons[same_ensembl, ] -->
<!-- exons <- exons[order(rownames(exons)), ] -->
<!-- introns <- introns[same_ensembl, ] -->
<!-- introns <- introns[order(rownames(introns)), ] -->
<!-- rownames(snrna_data) <- substring(rownames(snrna_data), 1, 18) -->
<!-- # Remove the introns and exons with the same Ensembl ID -->
<!-- # from the single-nuclear data. -->
<!-- snrna_data <- snrna_data[! rownames(snrna_data) %in% same_ensembl, ] -->
<!-- # A new SingleCellExperiment object was created, where the sum of the UMI counts -->
<!-- # of spliced mRNA and introns sharing the same Ensembl ID was added. -->
<!-- sce <- SingleCellExperiment(assays = list( -->
<!--   counts = rbind(counts(snrna_data), counts(exons) + counts(introns)) -->
<!-- )) -->
<!-- colData(sce) <- colData(snrna_data) -->
<!-- sce$sample_id[sce$sample_id == "8E"] <- "6E" -->
<!-- # We created the sleep condition variable -->
<!-- sce$condition <- sce$sample_id -->
<!-- sce$condition[grep("C", sce$condition)] <- "HC" # Home Cage (HC) -->
<!-- sce$condition[grep("E", sce$condition)] <- "SD" # Sleep Deprivated (SD) -->
<!-- sce$sample_id <- paste(substr(sce$sample_id, 1, 1), sce$condition, sep = "") -->
<!-- ``` -->
<!-- ```{r, eval=FALSE} -->
<!-- # To identify mitochondrial genes, we retrieved the chromosome location of -->
<!-- # each Ensembl Gene ID. -->
<!-- ensids <- rownames(sce) -->
<!-- map <- mapIds(EnsDb.Mmusculus.v79, keys = ensids, -->
<!--               column = "SEQNAME", keytype = "GENEID") -->
<!-- stopifnot(length(map) == nrow(sce)) -->
<!-- rowData(sce)$CHR <- map -->
<!-- # We split the data into six SingleCellExperiment objects, -->
<!-- # one for each mouse. -->
<!-- scelist <- list(sce[, sce$sample_id == "1HC"], sce[, sce$sample_id == "2SD"], -->
<!--                 sce[, sce$sample_id == "3HC"], sce[, sce$sample_id == "4SD"], -->
<!--                 sce[, sce$sample_id == "5HC"], sce[, sce$sample_id == "6SD"]) -->
<!-- ``` -->
<!-- ## Quality controls -->
<!-- For each sample, we used the Bioconductor scuttle package to detect low quality and damaged droplets. Particularly, we computed per-cell quality-control metrics with the *perCellQCMetrics* function; these metrics include the sum of UMI counts, the number of detected genes, and the percentage of mitochondrial counts.  -->
<!-- ```{r, eval=FALSE} -->
<!-- scelist_filt <- list() -->
<!-- for (i in seq_along(scelist)) { -->
<!--   stats <- perCellQCMetrics(scelist[[i]],  -->
<!--                             subsets = list(Mito = which(rowData(scelist[[i]])$CHR == "MT"))) -->
<!--   high_mito <- isOutlier(stats$subsets_Mito_percent, type = "higher") -->
<!--   colData(scelist[[i]]) <- cbind(colData(scelist[[i]]), stats) -->
<!--   scelist[[i]]$high_mito <- high_mito -->
<!--   qc_lib <- isOutlier(scelist[[i]]$sum, log = TRUE, type = "lower") -->
<!--   qc_nexprs <- isOutlier(scelist[[i]]$detected, log = TRUE, type = "lower") -->
<!--   discard <- qc_lib | qc_nexprs | scelist[[i]]$high_mito -->
<!--   scelist_filt[[i]] <- scelist[[i]][, !discard] -->
<!-- } -->
<!-- # Log-normalized counts -->
<!-- scelist_filt <- lapply(scelist_filt, function(x) logNormCounts(x)) -->
<!-- ``` -->
<!-- ## Doublets removal -->
<!-- Lastly, for each sample, we removed potential doublets with the scDblFinder (v1.12.0) package, using the *computeDoubletDensity* function to calculate the scores and the *doubletThresholding* function to set the doublet scores threshold with the *griffiths* method. -->
<!-- ```{r, eval=FALSE} -->
<!-- # HVGs were calculated for each sample -->
<!-- topgs <- lapply(scelist_filt, function(x) getTopHVGs(x, prop = 0.1)) -->
<!-- for (i in seq_along(scelist_filt)) { -->
<!--   set.seed(422) -->
<!--   # Function to calculate the scores -->
<!--   scores <- computeDoubletDensity(scelist_filt[[i]], subset.row = topgs[[i]]) -->
<!--   # Function to set the doublet scores threshold -->
<!--   dbl_calls <- doubletThresholding(data.frame(score = scores), method = "griffiths", returnType = "call") -->
<!--   colData(scelist_filt[[i]]) <- cbind.DataFrame(colData(scelist_filt[[i]]), dbl_calls, scores) -->
<!-- } -->
<!-- scelist_sgl <- lapply(scelist_filt, function(u) u[, !u$dbl_calls == "doublet"]) -->
<!-- names(scelist_sgl) <- levels(factor(sce$sample_id)) -->
<!-- scelist_sgl -->
<!-- ``` -->
<!-- # Cell-type annotation -->
<!-- ## Reference dataset -->
<!-- To identify cell types, we used the [Allen Whole Cortex & Hippocampus - 10x genomics (v2021)](https://www.sciencedirect.com/science/article/pii/S0092867421005018) as reference dataset . This dataset was imported by the *AllenInstituteBrainData* function of the AllenInstituteBrainData. We then selected the “Non-Neuronal”, “Glutamatergic” and “GABAergic” clusters coming from the Visual Cortex (VIS, VISl, VISm, VISp) to annotate our dataset. For computational issues, we selected a random subset of 100,000 cortical cells. -->
<!-- Cell annotation was computed with an automatic and reference-based method: Azimuth. -->
<!-- ```{r, eval=FALSE} -->
<!-- # Load reference dataset  -->
<!-- reference <- AllenInstituteBrainData("Allen_Mouse_2021") -->
<!-- rownames(reference) <- rowData(reference)$X -->
<!-- reference -->
<!-- # Select clusters from the Visual Cortex -->
<!-- reference <- reference[, c(reference$region_label == "VIS" | -->
<!--                              reference$region_label == "VISl" | -->
<!--                              reference$region_label == "VISm" | -->
<!--                              reference$region_label == "VISp")] -->
<!-- # Select cortex labels -->
<!-- reference <- reference[, !is.na(reference$subclass_label) & reference$subclass_label != ""] -->
<!-- reference <- reference[, -which(grepl("ENT", reference$subclass_label))] -->
<!-- reference <- reference[, -which(grepl("PPP", reference$subclass_label))] -->
<!-- reference <- reference[, -which(grepl("CR", reference$subclass_label))] -->
<!-- reference <- reference[, -which(grepl("Meis", reference$subclass_label))] -->
<!-- reference <- reference[, -which(grepl("SUB", reference$subclass_label))] -->
<!-- # For computational issues, we decided to select 100,000 cortical random cells. -->
<!-- # First, all cell was selected from the cell types with less than 100 cells. -->
<!-- no_random <- reference[, c(reference$subclass_label == "SMC-Peri" |  -->
<!--                              reference$subclass_label == "VLMC")] -->
<!-- # The resting cell types were selected. -->
<!-- reference <- reference[, !c(reference$subclass_label == "SMC-Peri" |  -->
<!--                               reference$subclass_label == "VLMC")] -->
<!-- df <- data.frame(colData(reference)) -->
<!-- # First, we randomly selected 100 cells for each cell type. -->
<!-- set.seed(23) -->
<!-- random <- df %>% group_by(subclass_label) %>% slice_sample(n = 100) -->
<!-- random100 <- reference[, colnames(reference) %in% random$sample_name] -->
<!-- reference <- reference[, !(colnames(reference) %in% random$sample_name)] -->
<!-- # Then, we randomly selected 98046 of the resting cells -->
<!-- set.seed(23) -->
<!-- reference <- reference[, sample(colnames(reference), 98046)] -->
<!-- reference <- cbind(reference, random100, no_random) -->
<!-- reference -->
<!-- ``` -->
<!-- ## Cell-type annotation with Azimuth  -->
<!-- For the Azimuth method, the reference data was converted into a Seurat object and into a Azimuth compatible object, using the *AzimuthReference* function of the Azimuth package. Then query samples were merged and were converted into a Seurat object.  -->
<!-- ```{r, eval=FALSE} -->
<!-- # First, the reference dataset was converted into Seurat object -->
<!-- counts <- as.matrix(counts(reference)) -->
<!-- coldata <- colData(reference) -->
<!-- reference_so <- CreateSeuratObject(counts = counts, -->
<!--                                    meta.data = data.frame(coldata)) -->
<!-- # And we created a compatible object for Azimuth cell annotation -->
<!-- reference_so <- SCTransform(reference_so, assay = "RNA", new.assay.name = "SCT", -->
<!--                             variable.features.n = 2000, verbose = TRUE, -->
<!--                             conserve.memory = TRUE) -->
<!-- reference_so <- RunPCA(reference_so, assay = "SCT", npcs = 50, verbose = FALSE, -->
<!--                        reduction.name = "PCA", return.model = TRUE) -->
<!-- reference_so <- RunUMAP(reference_so, assay = "SCT", reduction = "PCA", -->
<!--                         dims = seq_len(50), seed.use = 1, verbose = FALSE, -->
<!--                         reduction.name = "umap", return.model = TRUE) -->
<!-- reference_so$subclass_label <- as.factor(reference_so$subclass_label) -->
<!-- Idents(object = reference_so) <- "subclass_label" -->
<!-- # Azimuth-compatible object -->
<!-- reference_azimuth <- AzimuthReference(reference_so, refUMAP = "umap", -->
<!--                                       refDR = "PCA", refAssay = "SCT", -->
<!--                                       dims = 1:50, metadata = c("subclass_label"), -->
<!--                                       verbose = TRUE) -->
<!-- # save reference in a folder called "reference" -->
<!-- ref_dir <- "reference/" -->
<!-- SaveAnnoyIndex(object = reference_azimuth[["refdr.annoy.neighbors"]],  -->
<!--                file = file.path(ref_dir, "idx.annoy")) -->
<!-- saveRDS(object = reference_azimuth, file = file.path(ref_dir, "ref.Rds")) -->
<!-- ``` -->
<!-- The query dataset was converted into a Seurat object. -->
<!-- ```{r, eval=FALSE} -->
<!-- # The six SingleCellExperiment were combined and converted into a Seurat object. -->
<!-- counts <- cbind(counts(scelist_sgl[[1]]), counts(scelist_sgl[[2]]), -->
<!--                 counts(scelist_sgl[[3]]), counts(scelist_sgl[[4]]),  -->
<!--                 counts(scelist_sgl[[5]]), counts(scelist_sgl[[6]])) -->
<!-- sample_id <- c(scelist_sgl[[1]]$sample_id, scelist_sgl[[2]]$sample_id,  -->
<!--                scelist_sgl[[3]]$sample_id, scelist_sgl[[4]]$sample_id,  -->
<!--                scelist_sgl[[5]]$sample_id, scelist_sgl[[6]]$sample_id) -->
<!-- condition <- c(scelist_sgl[[1]]$condition, scelist_sgl[[2]]$condition,  -->
<!--                scelist_sgl[[3]]$condition, scelist_sgl[[4]]$condition,  -->
<!--                scelist_sgl[[5]]$condition, scelist_sgl[[6]]$condition) -->
<!-- sce_obj <-  SingleCellExperiment(assays = list(counts = counts)) -->
<!-- sce_obj$sample_id <- sample_id -->
<!-- sce_obj$condition <- condition -->
<!-- colnames(sce_obj) <- paste(colnames(sce_obj), sce_obj$sample_id, sep = "_") -->
<!-- # SCE object was converted into a Seurat object -->
<!-- seurat_obj <- CreateSeuratObject(counts = counts(sce_obj),  -->
<!--                                  meta.data = data.frame(colData(sce_obj))) -->
<!-- ``` -->
<!-- Cell annotation was computed using the *RunAzimuth* function of the Azimuth package. The t-SNE and the UMAP embeddings were computed using the *RunTSNE* and *RunUMAP* functions of the Seurat package with seed.use = 1. -->
<!-- ```{r, eval=FALSE} -->
<!-- # Cell-type annotation with Azimuth -->
<!-- seurat_obj <- RunAzimuth(seurat_obj, reference = "reference/") -->
<!-- # Dimensional reduction -->
<!-- seurat_obj <- RunTSNE(seurat_obj, reduction = "integrated_dr",  -->
<!--                       dims = seq_len(20), seed.use = 1, do.fast = TRUE,  -->
<!--                       verbose = FALSE, reduction.name = "TSNE") -->
<!-- seurat_obj <- RunUMAP(seurat_obj, reduction = "integrated_dr",  -->
<!--                       dims = seq_len(20), seed.use = 1, verbose = FALSE, -->
<!--                       reduction.name = "UMAP") -->
<!-- ``` -->
<!-- ```{r, echo=FALSE} -->
<!-- url <- "https://zenodo.org/records/10078998/files/snrna_seurat_annot.rds" -->
<!-- system(paste0("curl ", url)) -->
<!-- seurat_obj <- readRDS("snrna_seurat_annot.rds") -->
<!-- ``` -->
<!-- ## Dimensional reduction visualization  -->
<!-- To visualize the assigned labels in two dimensions, the UMAP embeddings were computed using the *DimPlot* function of the Seurat package, with option *reduction = “integrated_dr”*, where *“integrated_dr”* is the supervised principal component analysis obtained by the Azimuth method. -->
<!-- ```{r} -->
<!-- # Palette color creation  -->
<!-- label_color <- c("Astro", "Car3", "Endo", "L2/3 IT CTX", "L4/5 IT CTX", "L5 IT CTX", "L5 PT CTX", "L5/6 NP CTX", "L6 CT CTX", "L6 IT CTX" , "L6b CTX", "Lamp5", "Micro-PVM", "Oligo", "Pvalb",    "SMC-Peri", "Sncg", "Sst", "Sst Chodl", "Vip", "VLMC") -->
<!-- subclass_color <- c("#957b46", "#5100FF", "#c95f3f", "#0BE652", "#00E5E5", "#50B2AD", "#0D5B78",   "#3E9E64",  "#2D8CB8" ,  "#A19922",   "#7044AA",   "#DA808C", "#94AF97", "#744700", "#D93137",   "#4c1130" ,  "#ffff00",   "#FF9900", "#B1B10C",   "#B864CC",   "#a9bd4f") -->
<!-- names(subclass_color) <- label_color -->
<!-- umap_plot <- DimPlot(seurat_obj, reduction = "UMAP", group.by = "predicted.subclass_label")+ -->
<!--   NoLegend() + labs(x = "UMAP1", y="UMAP2") + ggtitle("") +  -->
<!--   scale_color_manual(values = subclass_color) -->
<!-- umap_plot <- LabelClusters(umap_plot, id = "predicted.subclass_label",  fontface = "bold", color = "black", size=2) -->
<!-- umap_plot <- umap_plot + theme(axis.text=element_text(size=7), axis.title=element_text(size=7)) -->
<!-- umap_plot -->
<!-- ``` -->
<!-- # Multidimensional Scaling (MDS) -->
<!-- For the next analysis, cell-types with more than 500 cells were selected. -->
<!-- ```{r} -->
<!-- # Add the Azimuth labels into the SingleCellExperiment object -->
<!-- sce_obj$azimuth_labels <- seurat_obj$predicted.subclass_label -->
<!-- # Cell-types with less than 500 cells were removed -->
<!-- sce_obj <- sce_obj[, !c(sce_obj$azimuth_labels == "Car3" | -->
<!--                           sce_obj$azimuth_labels == "Endo" | -->
<!--                           sce_obj$azimuth_labels == "Lamp5" | -->
<!--                           sce_obj$azimuth_labels == "Sncg" | -->
<!--                           sce_obj$azimuth_labels == "Sst Chodl" | -->
<!--                           sce_obj$azimuth_labels == "VLMC")] -->
<!-- ``` -->
<!-- We created the pseudo-bulk samples with the function *aggregateAcrossCells* of the scuttle package. In other words, we computed sum counts values for each feature for each cell-type and mouse. -->
<!-- ```{r} -->
<!-- # Pseudo-bulk creation -->
<!-- snrna_pb <- aggregateAcrossCells(sce_obj, use.assay.type = "counts",  -->
<!--                                  id = DataFrame(label = sce_obj$azimuth_labels,  -->
<!--                                                 sample = sce_obj$sample_id)) -->
<!-- colnames(snrna_pb) <- paste(snrna_pb$azimuth_labels, snrna_pb$sample_id, sep = "_") -->
<!-- snrna_pb <- logNormCounts(snrna_pb) -->
<!-- # Cell-type class variable was created -->
<!-- snrna_pb$class <- snrna_pb$azimuth_labels -->
<!-- snrna_pb$class[which(grepl("CTX", snrna_pb$class))] <- "Glutamatergic" -->
<!-- snrna_pb$class[which(grepl("-", snrna_pb$class))] <- "Other" -->
<!-- snrna_pb$class[!(snrna_pb$class == "Astro" | snrna_pb$class == "Oligo" | snrna_pb$class == "Glutamatergic" | -->
<!--              snrna_pb$class == "Other")] <- "GABAergic" -->
<!-- ``` -->
<!-- Finally, a pseudo-bulk level Multidimensional Scaling (MDS) plot was created with the *pbMDS* function of muscat package. Each point represents one subpopulation-sample instance; points are colored by subpopulation and shaped by treatment. -->
<!-- ```{r} -->
<!-- neuronal_color <- subclass_color[-c(1:3, 12:14, 16, 17, 19, 21)] -->
<!-- # MDS plot according to the Glutamatergic labels -->
<!-- prep_sce <- prepSCE(snrna_pb[, snrna_pb$class == "Glutamatergic"], -->
<!--                     kid = "azimuth_labels", -->
<!--                     gid = "condition",   -->
<!--                     sid = "sample_id", -->
<!--                     drop = TRUE) -->
<!-- pb <-  aggregateData(prep_sce, assay = "logcounts", -->
<!--                      by = c("cluster_id", "sample_id")) -->
<!-- pbMDS(pb) + scale_color_manual(values = neuronal_color[1:8]) + -->
<!--   labs(col = "Cell-types", shape = "Condition") + -->
<!--   theme(axis.text = element_text(size = 7), axis.title = element_text(size = 7)) -->
<!-- # MDS plot according to the GABA-ergic labels -->
<!-- prep_sce <- prepSCE(snrna_pb[, snrna_pb$class == "GABAergic"], -->
<!--                     kid = "azimuth_labels", gid = "condition", -->
<!--                     sid = "sample_id", drop = TRUE) -->
<!-- pb <-  aggregateData(prep_sce, assay = "logcounts", -->
<!--                      by = c("cluster_id", "sample_id")) -->
<!-- pbMDS(pb) + scale_color_manual(values = neuronal_color[9:11]) + -->
<!--   labs(col = "Cell-types", shape = "Condition") + -->
<!--   theme(axis.text = element_text(size = 7), axis.title = element_text(size = 7)) -->
<!-- # MDS plot according to the Non-Neuronal labels -->
<!-- prep_sce <- prepSCE(snrna_pb[, (snrna_pb$class == "Astro" | -->
<!--                                   snrna_pb$class == "Oligo" | -->
<!--                                   snrna_pb$class == "Other")], -->
<!--                     kid = "azimuth_labels", gid = "condition", -->
<!--                     sid = "sample_id", drop = TRUE) -->
<!-- pb <-  aggregateData(prep_sce, assay = "logcounts",  -->
<!--                      by = c("cluster_id", "sample_id")) -->
<!-- other_color <- subclass_color[c(1, 13:14, 16)] -->
<!-- pbMDS(pb) + scale_color_manual(values = other_color) + -->
<!--   labs(col = "Cell-types", shape = "Condition") + -->
<!--   theme(legend.spacing.y = unit(0, "cm"), axis.text = element_text(size = 7), -->
<!--         axis.title = element_text(size = 7), legend.title = element_text(size = 7), -->
<!--         legend.text = element_text(size = 7), legend.key.size = unit(0.4, "cm")) -->
<!-- # MDS plot according to the neuronal labels on negative control genes -->
<!-- # Load negative control genes -->
<!-- negctrl <- read.table("SD_Negative_Controls.txt") -->
<!-- # Negative control gene was selected inside the pseudo-bulk object -->
<!-- pb_negctrl <- snrna_pb[rownames(snrna_pb) %in% negctrl$x, ] -->
<!-- prep_sce <- prepSCE(pb_negctrl[, !(pb_negctrl$class == "Astro" | -->
<!--                                      pb_negctrl$class == "Oligo" | -->
<!--                                      pb_negctrl$class == "Other")], -->
<!--                     kid = "azimuth_labels", gid = "condition", -->
<!--                     sid = "sample", drop = TRUE) -->
<!-- pb <-  aggregateData(prep_sce, assay = "logcounts",  -->
<!--                      by = c("cluster_id", "sample_id")) -->
<!-- pbMDS(pb) + scale_color_manual(values = neuronal_color) + -->
<!--   labs(col = "Cell-types", shape = "Condition") + -->
<!--   theme(legend.spacing.y = unit(0, "cm"), axis.text = element_text(size = 7), -->
<!--         axis.title = element_text(size = 7), legend.title = element_text(size = 7), -->
<!--         legend.text = element_text(size = 7), legend.key.size = unit(0.4, "cm")) -->
<!-- ``` -->
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [7] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BiocStyle_2.30.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] knitr_1.45          magrittr_2.0.3      R6_2.5.1           </span></span>
<span><span class="co">##  [4] ragg_1.2.5          rlang_1.1.1         fastmap_1.1.1      </span></span>
<span><span class="co">##  [7] stringr_1.5.0       tools_4.2.3         xfun_0.40          </span></span>
<span><span class="co">## [10] cli_3.6.1           jquerylib_0.1.4     systemfonts_1.0.5  </span></span>
<span><span class="co">## [13] htmltools_0.5.6.1   yaml_2.3.7          digest_0.6.33      </span></span>
<span><span class="co">## [16] rprojroot_2.0.3     lifecycle_1.0.3     pkgdown_2.0.7      </span></span>
<span><span class="co">## [19] textshaping_0.3.7   formatR_1.14        BiocManager_1.30.22</span></span>
<span><span class="co">## [22] purrr_1.0.2         sass_0.4.7          vctrs_0.6.4        </span></span>
<span><span class="co">## [25] fs_1.6.3            memoise_2.0.1       glue_1.6.2         </span></span>
<span><span class="co">## [28] cachem_1.0.8        evaluate_0.22       rmarkdown_2.25     </span></span>
<span><span class="co">## [31] stringi_1.7.12      compiler_4.2.3      bslib_0.5.1        </span></span>
<span><span class="co">## [34] desc_1.4.2          jsonlite_1.8.7</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Elena Zuin, Dario Righelli, Davide Risso.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
