<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Bulk RNAseq Differential Expression Analysis • AtlasCortexSD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Bulk RNAseq Differential Expression Analysis">
<meta property="og:description" content="AtlasCortexSD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">AtlasCortexSD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_ct_anno.html">Cell Type Assignment</a>
    </li>
    <li>
      <a href="../articles/2_pb_dgea.html">Pseudo Bulk Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/3_bulk_de.html">Bulk RNAseq Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/4_Dtx_usage.html">Differential Transcripts Usage</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RissoLab/AtlasCortexSD" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Bulk RNAseq Differential Expression
Analysis</h1>
            
            <h4 data-toc-skip class="date">2023-11-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RissoLab/AtlasCortexSD/blob/HEAD/vignettes/3_bulk_de.Rmd" class="external-link"><code>vignettes/3_bulk_de.Rmd</code></a></small>
      <div class="hidden name"><code>3_bulk_de.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="differential-transcript-and-differential-gene-expression-using-fishpond-with-ruvs">Differential Transcript and Differential Gene Expression using
Fishpond with RUVs<a class="anchor" aria-label="anchor" href="#differential-transcript-and-differential-gene-expression-using-fishpond-with-ruvs"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>Sleep deprivation (SD) has negative effects on brain function. Sleep
problems are prevalent in neurodevelopmental, neurodegenerative and
psychiatric disorders. Thus, understanding the molecular consequences of
SD is of fundamental importance in neuroscience. In this study, we
present the first simultaneous bulk and single-nuclear (sn)RNA
sequencing characterization of the effects of SD in the mouse frontal
cortex. We show that SD predominantly affects glutamatergic neurons,
specifically in layers 4 and 5, and produces isoform switching of
thousands of transcripts. At both the global and cell-type specific
level, SD has a large repressive effect on transcription,
down-regulating thousands of genes and transcripts; underscoring the
importance of accounting for the effects of sleep loss in transcriptome
studies of brain function. As a resource we provide extensive
characterizations of cell types, genes, transcripts and pathways
affected by SD; as well as tutorials for data analysis.</p>
<p>This code was adapted from the tutorial here: <a href="https://bioconductor.org/packages/release/bioc/vignettes/fishpond/inst/doc/swish.html#Quick_start" class="external-link uri">https://bioconductor.org/packages/release/bioc/vignettes/fishpond/inst/doc/swish.html#Quick_start</a></p>
</div>
<div class="section level3">
<h3 id="data-availability">Data Availability<a class="anchor" aria-label="anchor" href="#data-availability"></a>
</h3>
<p>“Raw” sequencing data for this analysis is stored in GEO under
accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE211088" class="external-link">GSE211088</a>.
Additionally, the bulk files were previously published <span class="citation">[@ingiosi2019]</span> and can be accessed in GEO under
accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE113754" class="external-link">GSE113754</a>.</p>
<p>The code used in this analysis has been deposited into Github, and
can be accessed via the following link: <strong>link to github folder
where this code is</strong></p>
<p>Prior to this analysis, files were quantified using <em>Salmon</em>
<span class="citation">[@patro2017]</span> and imported into
<em>tximeta</em> <span class="citation">[@love2020]</span> to generate a
<em>SummarizedExperiment</em> object.</p>
</div>
<div class="section level3">
<h3 id="data-import">Data Import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h3>
<p>Establish the working directory using the code below. The working
directory can be thought of as the default location where files can be
accessed from or saved to.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="st">"~/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta"</span><span class="op">)</span></span>
<span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="st">"~/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta"</span><span class="op">)</span></span></code></pre></div>
<p>Import the colData, which contains the sample names and condition
(the condition is necessary for Fishpond). In short, the colData is a
file that contains descriptions of the samples, and is created during
<em>tximeta</em> <span class="citation">[@love2020]</span>. You can see
what the columns and first few rows of the file look like using the
<em>head()</em> function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"coldata.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                               names condition</span></span>
<span><span class="co">## S3HC5_PFC_1_quant S3HC5_PFC_1_quant     S3HC5</span></span>
<span><span class="co">## S3HC5_PFC_2_quant S3HC5_PFC_2_quant     S3HC5</span></span>
<span><span class="co">## S3HC5_PFC_3_quant S3HC5_PFC_3_quant     S3HC5</span></span>
<span><span class="co">## S3HC5_PFC_4_quant S3HC5_PFC_4_quant     S3HC5</span></span>
<span><span class="co">## S3HC5_PFC_5_quant S3HC5_PFC_5_quant     S3HC5</span></span>
<span><span class="co">## S3HC7_PFC_1_quant S3HC7_PFC_1_quant     S3HC7</span></span>
<span><span class="co">##                                                                                                                                        files</span></span>
<span><span class="co">## S3HC5_PFC_1_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC5_PFC_1_quant/quant.sf</span></span>
<span><span class="co">## S3HC5_PFC_2_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC5_PFC_2_quant/quant.sf</span></span>
<span><span class="co">## S3HC5_PFC_3_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC5_PFC_3_quant/quant.sf</span></span>
<span><span class="co">## S3HC5_PFC_4_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC5_PFC_4_quant/quant.sf</span></span>
<span><span class="co">## S3HC5_PFC_5_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC5_PFC_5_quant/quant.sf</span></span>
<span><span class="co">## S3HC7_PFC_1_quant /Users/kaitlynford/Dropbox/Sleep_RNAseq_Splicing/BulkRNAseq_SD/Final_Tximeta/data/Salmon_Quants/S3HC7_PFC_1_quant/quant.sf</span></span></code></pre>
<p>Add a path to locate the files:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coldata</span><span class="op">$</span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"data/Salmon_Quants"</span>, <span class="va">coldata</span><span class="op">$</span><span class="va">names</span>,</span>
<span>    <span class="st">"quant.sf"</span><span class="op">)</span></span></code></pre></div>
<p>Before continuing with the analysis, make sure that all files in your
colData exist in this location:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">$</span><span class="va">files</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p>Load <em>SummarizedExperiment</em> and <em>tximeta</em> packages:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 1.28.0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tximeta</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 1.16.1</span></span></code></pre></div>
<p>Load the quantification data with <em>tximeta</em> (Here we load with
an rda file):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"se.rda"</span><span class="op">)</span></span></code></pre></div>
<p>Ensure assays are loaded by viewing <em>assayNames()</em>. Assays are
experimental results, and are necessary for <em>fishpond</em>. Call
<em>assay()</em> to retrieve experimental data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "counts"    "abundance" "length"    "infRep1"   "infRep2"  </span></span>
<span><span class="co">##  [6] "infRep3"   "infRep4"   "infRep5"   "infRep6"   "infRep7"  </span></span>
<span><span class="co">## [11] "infRep8"   "infRep9"   "infRep10"  "infRep11"  "infRep12" </span></span>
<span><span class="co">## [16] "infRep13"  "infRep14"  "infRep15"  "infRep16"  "infRep17" </span></span>
<span><span class="co">## [21] "infRep18"  "infRep19"  "infRep20"  "infRep21"  "infRep22" </span></span>
<span><span class="co">## [26] "infRep23"  "infRep24"  "infRep25"  "infRep26"  "infRep27" </span></span>
<span><span class="co">## [31] "infRep28"  "infRep29"  "infRep30"</span></span></code></pre>
<p>View rownames (transcript IDs) using the <em>head()</em>
function:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSMUST00000193812.1" "ENSMUST00000082908.1"</span></span>
<span><span class="co">## [3] "ENSMUST00000162897.1" "ENSMUST00000159265.1"</span></span>
<span><span class="co">## [5] "ENSMUST00000070533.4" "ENSMUST00000192857.1"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="differential-transcript-expression-analysis-starts-here">Differential Transcript Expression Analysis Starts Here<a class="anchor" aria-label="anchor" href="#differential-transcript-expression-analysis-starts-here"></a>
</h2>
<p>‘se’ will be referred to as ‘y’ for the remainder of the
analysis.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">se</span></span></code></pre></div>
<p>I will do a two group comparison between wild-type animals. Those
that were left undisturbed in their home cage (HC) for 5 hours and those
who were sleep sleep deprived (SD) via gentle handling for 5 hours.</p>
<p>The following analysis will compare the second group to the first
group listed in <em>levels</em>, in this case, SD compared to HC
controls.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">y</span><span class="op">[</span>, <span class="va">y</span><span class="op">$</span><span class="va">condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WTHC5"</span>, <span class="st">"WTSD5"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">y</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">y</span><span class="op">$</span><span class="va">condition</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WTHC5"</span>, <span class="st">"WTSD5"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-fishpond-and-proceed-with-differential-expression-analysis">Load Fishpond, and Proceed with Differential Expression
Analysis<a class="anchor" aria-label="anchor" href="#load-fishpond-and-proceed-with-differential-expression-analysis"></a>
</h3>
<p>Load the <em>fishpond</em> package to proceed with differential
expression analysis:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">fishpond</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 2.4.1</span></span></code></pre></div>
<p>Scale inferential replicates to the mean sequencing depth. It is
important to note that only the inferential replicates are scaled during
this process, not the counts, which will be relevant in later steps.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">scaleInfReps</span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Progress:  1 on 30  Progress:  2 on 30  Progress:  3 on 30  Progress:  4 on 30  Progress:  5 on 30  Progress:  6 on 30  Progress:  7 on 30  Progress:  8 on 30  Progress:  9 on 30  Progress: 10 on 30  Progress: 11 on 30  Progress: 12 on 30  Progress: 13 on 30  Progress: 14 on 30  Progress: 15 on 30  Progress: 16 on 30  Progress: 17 on 30  Progress: 18 on 30  Progress: 19 on 30  Progress: 20 on 30  Progress: 21 on 30  Progress: 22 on 30  Progress: 23 on 30  Progress: 24 on 30  Progress: 25 on 30  Progress: 26 on 30  Progress: 27 on 30  Progress: 28 on 30  Progress: 29 on 30  Progress: 30 on 30</span></span></code></pre>
<p>Before filtering, we have 140,992 transcripts:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 140992     10</span></span></code></pre>
<p>Filter here, keeping a minimum count of 10 transcripts across a
minimum of 3 replicates. In this step, we are removing most of the rows
with zero to very low counts before proceeding with the analysis.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">labelKeep</span><span class="op">(</span><span class="va">y</span>, minCount <span class="op">=</span> <span class="fl">10</span>, minN <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p>After filtering, we have 54,030 transcripts:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">y</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">$</span><span class="va">keep</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 54030    10</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="determining-the-sources-of-unwanted-variation-with-ruvs">Determining the Sources of Unwanted Variation with RUVs<a class="anchor" aria-label="anchor" href="#determining-the-sources-of-unwanted-variation-with-ruvs"></a>
</h3>
<p>Assemble a matrix that contains the groups. If you have uneven
groups, “fill” the extra spaces in the matrix with “-1” values.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Check to make sure ‘groups’ appears as expected.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2] [,3] [,4] [,5]</span></span>
<span><span class="co">## [1,]    1    2    3    4    5</span></span>
<span><span class="co">## [2,]    6    7    8    9   10</span></span></code></pre>
<p>Assign the names of the variables to match the column names of your
data. This is important for <em>RColorBrewer</em>, which is used to
designate colors for figures later on in this analysis:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WTHC5"</span>, <span class="st">"WTSD5"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] WTHC5 WTHC5 WTHC5 WTHC5 WTHC5 WTSD5 WTSD5 WTSD5 WTSD5 WTSD5</span></span>
<span><span class="co">## Levels: WTHC5 WTSD5</span></span></code></pre>
<p>Shorten sample names for simplicity while making the matrix:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HC5_1"</span>, <span class="st">"HC5_2"</span>, <span class="st">"HC5_3"</span>, <span class="st">"HC5_4"</span>, <span class="st">"HC5_5"</span>, <span class="st">"SD5_1"</span>,</span>
<span>    <span class="st">"SD5_2"</span>, <span class="st">"SD5_3"</span>, <span class="st">"SD5_4"</span>, <span class="st">"SD5_5"</span><span class="op">)</span></span></code></pre></div>
<p>See what type of data class ‘x’ is using the code below:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "factor"</span></span></code></pre>
<p>Turn ‘x’ into a matrix:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       [,1]   </span></span>
<span><span class="co">## HC5_1 "WTHC5"</span></span>
<span><span class="co">## HC5_2 "WTHC5"</span></span>
<span><span class="co">## HC5_3 "WTHC5"</span></span>
<span><span class="co">## HC5_4 "WTHC5"</span></span>
<span><span class="co">## HC5_5 "WTHC5"</span></span>
<span><span class="co">## SD5_1 "WTSD5"</span></span>
<span><span class="co">## SD5_2 "WTSD5"</span></span>
<span><span class="co">## SD5_3 "WTSD5"</span></span>
<span><span class="co">## SD5_4 "WTSD5"</span></span>
<span><span class="co">## SD5_5 "WTSD5"</span></span></code></pre>
<p>Load <em>RColorBrewer</em> here:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 1.1-3</span></span></code></pre></div>
<p>This palette will be used for PCA and RLE plots for both transcript
and gene expression analysis. HC controls will be light grey and SD
samples will be dark grey. To view other palette options, use
<em>brewer.pal.info</em>:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Greys"</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>, <span class="fl">7</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">colLib</span> <span class="op">&lt;-</span> <span class="va">colors</span><span class="op">[</span><span class="va">x</span><span class="op">]</span></span>
<span><span class="va">HC_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Greys"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="va">SD_color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Greys"</span><span class="op">)</span><span class="op">[</span><span class="fl">7</span><span class="op">]</span></span></code></pre></div>
<p>Remove the version #s for each transcript. It is helpful to note that
the version #s correspond to the release version, they are not different
transcripts of the same gene:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, <span class="va">sub</span>, pattern <span class="op">=</span> <span class="st">"\\.\\d+$"</span>,</span>
<span>    replacement <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "character"</span></span></code></pre>
<p>Extract the counts from the <em>SummarizedExperiment</em> object for
normalization. Note that we will estimate the factors of unwanted
variation on the counts, and then we will apply that across the
inferential replicates further downstream:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">[[</span><span class="st">"counts"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "matrix"</span></span></code></pre>
<p>Upper Quartile (UQ) normalization is implemented here with the
<em>EDASeq</em> package, and accounts for variation in sequencing
depth:</p>
<p>As the counts were not scaled (only the inferential replicates), we
need to account for sequencing depth prior to proceeding with
<em>RUVs</em> normalization, which allows us to obtain a the estimated
factors of unwanted variation (<em>s$W</em>):</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/drisso/EDASeq" class="external-link">"EDASeq"</a></span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 2.32.0</span></span>
<span><span class="va">uq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html" class="external-link">betweenLaneNormalization</a></span><span class="op">(</span><span class="va">counts</span>, which <span class="op">=</span> <span class="st">"upper"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">uq</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 54030    10</span></span></code></pre>
<p>Shorten the sample names (<em>colnames</em>) for figures:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">uq</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "WTHC5_PFC_1_quant" "WTHC5_PFC_2_quant" "WTHC5_PFC_3_quant"</span></span>
<span><span class="co">##  [4] "WTHC5_PFC_4_quant" "WTHC5_PFC_5_quant" "WTSD5_PFC_1_quant"</span></span>
<span><span class="co">##  [7] "WTSD5_PFC_2_quant" "WTSD5_PFC_3_quant" "WTSD5_PFC_4_quant"</span></span>
<span><span class="co">## [10] "WTSD5_PFC_5_quant"</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">uq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HC5_1"</span>, <span class="st">"HC5_2"</span>, <span class="st">"HC5_3"</span>, <span class="st">"HC5_4"</span>, <span class="st">"HC5_5"</span>,</span>
<span>    <span class="st">"SD5_1"</span>, <span class="st">"SD5_2"</span>, <span class="st">"SD5_3"</span>, <span class="st">"SD5_4"</span>, <span class="st">"SD5_5"</span><span class="op">)</span></span></code></pre></div>
<p>Here we plot two quality control plots (RLE and PCA) following UQ
normalization. Note that triangles are sleep deprived animals and
circles are home cage control animals that were collected at the same
circadian time:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="va">uq</span>, col <span class="op">=</span> <span class="va">colLib</span>, outline <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">3</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,</span>
<span>    <span class="fl">1</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Relative Log Expression"</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, cex.lab <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-28-1.png" alt="plot of chunk unnamed-chunk-28" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">uq</span>, labels <span class="op">=</span> <span class="cn">FALSE</span>, col <span class="op">=</span> <span class="va">colLib</span>, cex <span class="op">=</span> <span class="fl">1.5</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>,</span>
<span>    cex.lab <span class="op">=</span> <span class="fl">1.3</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>    pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">19</span>, <span class="fl">17</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-28-2.png" alt="plot of chunk unnamed-chunk-28" style="display: block; margin: auto;"></p>
<p>Load <em>RUVSeq</em> package:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drisso/RUVSeq" class="external-link">RUVSeq</a></span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 1.32.0</span></span></code></pre></div>
<p><em>RUVseq</em>: Here we will estimate a matrix that contains
estimated factors of unwanted factors after UQ normalization.</p>
<p><em>RUVs</em> uses technical replicates or negative controls samples.
In this instance, as we do not have negative controls for transcripts,
we will use the rownames of the matrix y after filtering (all expressed
transcripts). For additional information regarding <em>RUVs</em>, see
<span class="citation">[@risso2014]</span>.</p>
<p>We have set the numbers of factors of unwanted variation (<em>k</em>)
= 4, which optimizes the positive control recovery (at the gene level)
without removing signal. When one is beginning analysis, it is
recommended to test a variety of <em>k</em>s, and compare the results to
one another.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RUVSeq/man/RUVs.html" class="external-link">RUVs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">uq</span>, cIdx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, scIdx <span class="op">=</span> <span class="va">groups</span>, k <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p>Plot the RLE and PCA plots (quality control figures) again following
<em>RUVs</em> normalization. After proper normalization, the largest
source of variation in the dataset is the treatment as visualized on the
PCA plot (separated along PC1).</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="va">s</span><span class="op">$</span><span class="va">normalizedCounts</span>, col <span class="op">=</span> <span class="va">colLib</span>, outline <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Relative Log Expression"</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    cex.lab <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-31-1.png" alt="plot of chunk unnamed-chunk-31" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">s</span><span class="op">$</span><span class="va">normalizedCounts</span>, labels <span class="op">=</span> <span class="cn">FALSE</span>, col <span class="op">=</span> <span class="va">colLib</span>, cex <span class="op">=</span> <span class="fl">1.3</span>,</span>
<span>    cex.axis <span class="op">=</span> <span class="fl">1.2</span>, cex.lab <span class="op">=</span> <span class="fl">1.2</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>,</span>
<span>        <span class="fl">0.75</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">19</span>, <span class="fl">17</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-31-2.png" alt="plot of chunk unnamed-chunk-31" style="display: block; margin: auto;"></p>
<p>Downstream we will use <em>s$W</em> (from <em>RUVs</em>) as the
estimation of batch factors.</p>
<p>‘infRepIdx’ lists the inferential replicates (30):</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infRepIdx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"infRep"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Save the number of inferential replicates as ‘nreps’:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nreps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">infRepIdx</span><span class="op">)</span></span>
<span><span class="va">nreps</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 30</span></span></code></pre>
<p>Account for continuous variables with <em>removeBatchEffect</em> from
<em>limma</em>. Our samples do not fall into discrete clusters so we
will use the following approach recommended by the <em>fishpond</em>
developers.</p>
<p>In short, this is done by directly scaling the estimated counts
across inferential replicates.</p>
<p>First inferential replicates are logged as <em>limma</em> requires
log-expression values for a series of samples. This is done with the
assay function from the <em>SummarizedExperiment</em> package.</p>
<p>First, we load the <em>limma</em> package:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span><span class="op">)</span>  <span class="co"># Version 3.54.2</span></span></code></pre></div>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">condition</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pc</span> <span class="op">&lt;-</span> <span class="fl">0.1</span>  <span class="co"># This is added to avoid negative InfReps</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">nreps</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">logInfRep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">infRepIdx</span><span class="op">[</span><span class="va">k</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="va">pc</span><span class="op">)</span></span>
<span>    <span class="va">logInfRep</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect</a></span><span class="op">(</span><span class="va">logInfRep</span>, covariates <span class="op">=</span> <span class="va">s</span><span class="op">$</span><span class="va">W</span>,</span>
<span>        design <span class="op">=</span> <span class="va">model_matrix</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">infRepIdx</span><span class="op">[</span><span class="va">k</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="va">logInfRep</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-swish-and-plot-additional-figures">Run Swish and Plot Additional Figures<a class="anchor" aria-label="anchor" href="#run-swish-and-plot-additional-figures"></a>
</h3>
<p>The Swish method is described in <span class="citation">[@zhu2019]</span>. Note that the <em>set.seed</em>
function allows for the reproducibility of exact results in the
future.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">swish</span><span class="op">(</span><span class="va">y</span>, x <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100</span></span></code></pre>
<p>View a table of differently expressed transcripts with a qvalue &lt;
0.05:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">$</span><span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 38505 15525</span></span></code></pre>
<p>You can also view a table that shows which transcripts are up
regulated (1) in response to sleep deprivation and which are
downregulated (-1):</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, sign.lfc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        sign.lfc</span></span>
<span><span class="co">## sig        -1     0     1</span></span>
<span><span class="co">##   FALSE 25479    11 13015</span></span>
<span><span class="co">##   TRUE   9709     0  5816</span></span></code></pre>
<p>Here, we will export a text file (containing the median of each
transcript across inferential replicates) for additional plotting.
First, select all features, use abind to combine multi-dimensional
arrays, and then take the median:</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infReps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"infRep"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">infArray</span> <span class="op">&lt;-</span> <span class="fu">abind</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/abind/man/abind.html" class="external-link">abind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">infReps</span><span class="op">)</span>, along <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">infArray</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 54030    10    30</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infMed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">infArray</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">median</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="va">infMed</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "matrix"</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">infMed</span>, file <span class="op">=</span> <span class="st">"DTE_InfMed.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div>
<p>Plot a histogram to show the distribution of pvalues:</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">$</span><span class="va">pvalue</span>, col <span class="op">=</span> <span class="st">"grey"</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">25000</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">""</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"Pvalue"</span>, cex.axis <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-40-1.png" alt="plot of chunk unnamed-chunk-40" style="display: block; margin: auto;"></p>
<p>Add symbols (Gene IDs) to the <em>SummarizedExperiment</em> object.
We will label genes on the MA plot in later steps. First load the
<em>org.Mm.eg.db</em> package:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span><span class="op">)</span>  <span class="co">#Version 3.16.0</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">addIds</span><span class="op">(</span><span class="va">y</span>, <span class="st">"SYMBOL"</span>, gene <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 54030 rows and 11 columns</span></span>
<span><span class="co">##                        tx_id              gene_id</span></span>
<span><span class="co">##                    &lt;integer&gt;      &lt;CharacterList&gt;</span></span>
<span><span class="co">## ENSMUST00000162897      4224 ENSMUSG00000051951.5</span></span>
<span><span class="co">## ENSMUST00000159265      4225 ENSMUSG00000051951.5</span></span>
<span><span class="co">## ENSMUST00000070533      4226 ENSMUSG00000051951.5</span></span>
<span><span class="co">## ENSMUST00000192692      4232 ENSMUSG00000102331.1</span></span>
<span><span class="co">## ENSMUST00000193244         6 ENSMUSG00000102348.1</span></span>
<span><span class="co">## ...                      ...                  ...</span></span>
<span><span class="co">## ENSMUST00000084013    142588 ENSMUSG00000065947.3</span></span>
<span><span class="co">## ENSMUST00000082414    142589 ENSMUSG00000064363.1</span></span>
<span><span class="co">## ENSMUST00000082418    142593 ENSMUSG00000064367.1</span></span>
<span><span class="co">## ENSMUST00000082419    142602 ENSMUSG00000064368.1</span></span>
<span><span class="co">## ENSMUST00000082421    142594 ENSMUSG00000064370.1</span></span>
<span><span class="co">##                                 tx_name log10mean      keep</span></span>
<span><span class="co">##                             &lt;character&gt; &lt;numeric&gt; &lt;logical&gt;</span></span>
<span><span class="co">## ENSMUST00000162897 ENSMUST00000162897.1   2.04050      TRUE</span></span>
<span><span class="co">## ENSMUST00000159265 ENSMUST00000159265.1   1.70889      TRUE</span></span>
<span><span class="co">## ENSMUST00000070533 ENSMUST00000070533.4   1.82758      TRUE</span></span>
<span><span class="co">## ENSMUST00000192692 ENSMUST00000192692.1   1.72937      TRUE</span></span>
<span><span class="co">## ENSMUST00000193244 ENSMUST00000193244.1   1.16536      TRUE</span></span>
<span><span class="co">## ...                                 ...       ...       ...</span></span>
<span><span class="co">## ENSMUST00000084013 ENSMUST00000084013.1   4.01172      TRUE</span></span>
<span><span class="co">## ENSMUST00000082414 ENSMUST00000082414.1   5.48958      TRUE</span></span>
<span><span class="co">## ENSMUST00000082418 ENSMUST00000082418.1   5.36163      TRUE</span></span>
<span><span class="co">## ENSMUST00000082419 ENSMUST00000082419.1   3.24554      TRUE</span></span>
<span><span class="co">## ENSMUST00000082421 ENSMUST00000082421.1   5.58889      TRUE</span></span>
<span><span class="co">##                         stat     log2FC    pvalue    locfdr</span></span>
<span><span class="co">##                    &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## ENSMUST00000162897  -7.73333  -0.268533 0.0420622  0.375400</span></span>
<span><span class="co">## ENSMUST00000159265   3.66667   0.106852 0.3208627  1.000000</span></span>
<span><span class="co">## ENSMUST00000070533  -4.96667  -0.218354 0.1850087  1.000000</span></span>
<span><span class="co">## ENSMUST00000192692  -6.00000  -0.226598 0.1135445  0.763961</span></span>
<span><span class="co">## ENSMUST00000193244   6.66667   0.219371 0.0800030  0.599861</span></span>
<span><span class="co">## ...                      ...        ...       ...       ...</span></span>
<span><span class="co">## ENSMUST00000084013   1.96667  0.0564183  0.597025         1</span></span>
<span><span class="co">## ENSMUST00000082414   0.50000  0.0194699  0.892375         1</span></span>
<span><span class="co">## ENSMUST00000082418  -3.50000 -0.0413981  0.343077         1</span></span>
<span><span class="co">## ENSMUST00000082419  -4.30000 -0.1096372  0.246971         1</span></span>
<span><span class="co">## ENSMUST00000082421  -2.53333 -0.0165749  0.493297         1</span></span>
<span><span class="co">##                       qvalue      SYMBOL</span></span>
<span><span class="co">##                    &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">## ENSMUST00000162897  0.106326        Xkr4</span></span>
<span><span class="co">## ENSMUST00000159265  0.456228        Xkr4</span></span>
<span><span class="co">## ENSMUST00000070533  0.306843        Xkr4</span></span>
<span><span class="co">## ENSMUST00000192692  0.217177          NA</span></span>
<span><span class="co">## ENSMUST00000193244  0.169147          NA</span></span>
<span><span class="co">## ...                      ...         ...</span></span>
<span><span class="co">## ENSMUST00000084013  0.712552        ND4L</span></span>
<span><span class="co">## ENSMUST00000082414  0.930020         ND4</span></span>
<span><span class="co">## ENSMUST00000082418  0.478829         ND5</span></span>
<span><span class="co">## ENSMUST00000082419  0.377180         ND6</span></span>
<span><span class="co">## ENSMUST00000082421  0.621844        CYTB</span></span></code></pre>
<p>Turn ‘y’ into a dataframe here, and save as a different variable. We
do this to intersect with other lists when making plots, such as genes
to highlight on the MA plot downstream. Check the <em>dim</em> to ensure
that the size of the data frame has not changed:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y_rowData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">y_rowData</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 54030    11</span></span></code></pre>
<p>Here we graph an MA plot. An MA plot shows log2FC vs the log10mean.
Transcripts above M=0 are upregulated, while transcripts below are
downregulated.</p>
<p>The transcripts that are farthest away from the y intercept (have the
largest log10mean values) are the ones most affected by the
treatment.</p>
<p>Set colors for the MA plot here: Transcripts that do not have a
significant change in expression are light grey. Transcripts that are
significantly affected (qvalue &lt; 0.05) by SD are dark grey.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Significant_Color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Greys"</span><span class="op">)</span><span class="op">[</span><span class="fl">8</span><span class="op">]</span></span></code></pre></div>
<p>A subset of transcripts that we have chosen to highlight that also
have significant changes in expression (qvalue &lt; 0.05) will be bright
blue. Downstream, we will apply an abs(log2FC) threshold of 0.2 to these
transcripts.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Highlight_Color</span> <span class="op">&lt;-</span> <span class="st">"dodgerblue"</span></span></code></pre></div>
<p>Read the list of genes that we have chosen to highlight here.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"Highlight_Genes.rda"</span><span class="op">)</span></span>
<span><span class="va">Highlight_Genes</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 11 × 1</span></span>
<span><span class="co">##    ...1    </span></span>
<span><span class="co">##    &lt;chr&gt;   </span></span>
<span><span class="co">##  1 Arc     </span></span>
<span><span class="co">##  2 Bdnf    </span></span>
<span><span class="co">##  3 Cirbp   </span></span>
<span><span class="co">##  4 Eif4ebp1</span></span>
<span><span class="co">##  5 Fos     </span></span>
<span><span class="co">##  6 Homer1  </span></span>
<span><span class="co">##  7 Mef2c   </span></span>
<span><span class="co">##  8 Hdac7   </span></span>
<span><span class="co">##  9 Hspa5   </span></span>
<span><span class="co">## 10 Sst     </span></span>
<span><span class="co">## 11 Wnt9a</span></span></code></pre>
<p>Save ‘Highlight_Genes’ as a matrix titled ‘Gene_Names’:</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Gene_Names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">Highlight_Genes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Gene_Names</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 11  1</span></span></code></pre>
<p>Remove commas, quotes and trailing spaces:</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Gene_Names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">Gene_Names</span>, <span class="st">","</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Gene_Names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/noquote.html" class="external-link">noquote</a></span><span class="op">(</span><span class="va">Gene_Names</span><span class="op">)</span></span>
<span><span class="va">Gene_Names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/trimws.html" class="external-link">trimws</a></span><span class="op">(</span><span class="va">Gene_Names</span><span class="op">)</span></span></code></pre></div>
<p>We will intersect ‘y_rowData’ (the dataframe saved above) with the
‘Gene_Names’ matrix just saved using the <em>filter</em> function of
<em>dplyr</em>:</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Data_Intersection</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">y_rowData</span>, <span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">Gene_Names</span> <span class="op">&amp;</span></span>
<span>    <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 33 11</span></span></code></pre>
<p>Use <em>plotMASwish</em> to generate the MA plot. Include labels and
points for the genes in ‘Gene_Names’ using the code below (Only
transcripts that fit the qvalue and log2FC threshold will be
labeled):</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMASwish</span><span class="op">(</span><span class="va">y</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">6</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">8</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>    sigcolor <span class="op">=</span> <span class="va">Significant_Color</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, cex.lab <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, <span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">Gene_Names</span> <span class="op">&amp;</span> <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="va">log10mean</span>, <span class="va">log2FC</span>, <span class="va">SYMBOL</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>,</span>
<span>    pos <span class="op">=</span> <span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">0.8</span>, font <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log10mean</span>, <span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log2FC</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>, cex <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log10mean</span>, <span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log2FC</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">20</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>, cex <span class="op">=</span> <span class="fl">0.2</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-49-1.png" alt="plot of chunk unnamed-chunk-49" style="display: block; margin: auto;"></p>
<p>Write text file using the code below. Select columns to include in
the text file. These lists will then be annotated in Perl with the gene
name and description.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"log10mean"</span>, <span class="st">"log2FC"</span>, <span class="st">"pvalue"</span>, <span class="st">"qvalue"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="va">y_filter</span> <span class="op">&lt;-</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">y_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">y_filter</span><span class="op">)</span><span class="op">[</span>, <span class="va">cols</span><span class="op">]</span></span>
<span><span class="va">y_filter</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">y_filter</span>, <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">y_filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 15525     5</span></span></code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">y_filter</span>, file <span class="op">=</span> <span class="st">"DTE_Significant_k=4.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="differential-gene-expression-starts-here">Differential Gene Expression Starts Here<a class="anchor" aria-label="anchor" href="#differential-gene-expression-starts-here"></a>
</h2>
<p>Use the <em>summarizeToGene</em> function to proceed with analysis at
the gene level. For the remainder of the analysis, ‘gse’ will be
referred to as ‘gy’:</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="va">gse</span></span></code></pre></div>
<p>I will do the same two group comparison between wild-type animals
that was done at the transcript level. To quickly summarize this again,
we will compare animals that were left undisturbed in their home cage
(HC) for 5 hours and those who were sleep sleep deprived (SD) via gentle
handling for 5 hours.</p>
<p>As above, the following analysis will compare the second group to the
first group listed in <em>levels</em>, in this case, SD compared to HC
controls.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="va">gy</span><span class="op">[</span>, <span class="va">gy</span><span class="op">$</span><span class="va">condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WTHC5"</span>, <span class="st">"WTSD5"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">gy</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">gy</span><span class="op">$</span><span class="va">condition</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WTHC5"</span>, <span class="st">"WTSD5"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="proceed-with-differential-expression-analysis">Proceed with Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#proceed-with-differential-expression-analysis"></a>
</h3>
<p>Scale inferential replicates to the mean sequencing depth. It is
important to note that only the inferential replicates are scaled during
this process, not the counts.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">scaleInfReps</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Progress:  1 on 30  Progress:  2 on 30  Progress:  3 on 30  Progress:  4 on 30  Progress:  5 on 30  Progress:  6 on 30  Progress:  7 on 30  Progress:  8 on 30  Progress:  9 on 30  Progress: 10 on 30  Progress: 11 on 30  Progress: 12 on 30  Progress: 13 on 30  Progress: 14 on 30  Progress: 15 on 30  Progress: 16 on 30  Progress: 17 on 30  Progress: 18 on 30  Progress: 19 on 30  Progress: 20 on 30  Progress: 21 on 30  Progress: 22 on 30  Progress: 23 on 30  Progress: 24 on 30  Progress: 25 on 30  Progress: 26 on 30  Progress: 27 on 30  Progress: 28 on 30  Progress: 29 on 30  Progress: 30 on 30</span></span></code></pre>
<p>Before filtering, we have 54,347 genes:</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 54347    10</span></span></code></pre>
<p>Filter here, keeping a minimum count of 10 transcripts across a
minimum of 3 replicates. In this step, we are removing most of the rows
with zero to very low counts before proceeding with the analysis.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">labelKeep</span><span class="op">(</span><span class="va">gy</span>, minCount <span class="op">=</span> <span class="fl">10</span>, minN <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p>After filtering, we have 18,334 genes:</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="va">gy</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">$</span><span class="va">keep</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 18334    10</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="determining-the-sources-of-unwanted-variation-with-ruvs-1">Determining the Sources of Unwanted Variation with RUVs<a class="anchor" aria-label="anchor" href="#determining-the-sources-of-unwanted-variation-with-ruvs-1"></a>
</h3>
<p>During transcript expression analysis we already assembled a matrix
that contains the groups. View it here to make sure ‘groups’ appears as
expected.</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2] [,3] [,4] [,5]</span></span>
<span><span class="co">## [1,]    1    2    3    4    5</span></span>
<span><span class="co">## [2,]    6    7    8    9   10</span></span></code></pre>
<p>We will use the same colors that were used for differential
transcript expression, so this step does not need to be repeated. Again,
HC controls will be light grey and SD samples will be dark grey.</p>
<p>Remove the version #s for each gene. It is helpful to note that the
version #s correspond to the release version:</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span>, <span class="va">sub</span>, pattern <span class="op">=</span> <span class="st">"\\.\\d+$"</span>,</span>
<span>    replacement <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "character"</span></span></code></pre>
<p>Extract the counts from the <em>SummarizedExperiment</em> object for
normalization. Note that we will estimate the factors of unwanted
variation on the counts, and then we will apply that across the
inferential replicates further downstream:</p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts_gy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">[[</span><span class="st">"counts"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.class.html" class="external-link">data.class</a></span><span class="op">(</span><span class="va">counts_gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "matrix"</span></span></code></pre>
<p>For gene level analysis, we can use a set of positive controls from
<span class="citation">[@gerstner2016]</span> to assess the performance
of our pipeline downstream. Load the positive controls here:</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"Gene_Positive_Controls.rda"</span><span class="op">)</span></span></code></pre></div>
<p>In preparation for determining our positive control recovery, we will
then want to intersect the expressed genes with the positive
controls.</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intersect_Positive_Expressed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">Gene_Positive_Controls</span><span class="op">[</span>,</span>
<span>    <span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">counts_gy</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">Intersect_Positive_Expressed</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 671</span></span></code></pre>
<p>Upper Quartile (UQ) normalization is implemented here with the
<em>EDASeq</em> package (same as with transcript expression analysis),
and accounts for variation in sequencing depth:</p>
<p>As the counts were not scaled (only the inferential replicates), we
need to account for sequencing depth prior to proceeding with
<em>RUVs</em> normalization, which allows us to obtain a the estimated
factors of unwanted variation (<em>gs$W</em>):</p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g_uq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html" class="external-link">betweenLaneNormalization</a></span><span class="op">(</span><span class="va">counts_gy</span>, which <span class="op">=</span> <span class="st">"upper"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">g_uq</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 18334    10</span></span></code></pre>
<p>Shorten the sample names (<em>colnames</em>) for figures:</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">g_uq</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "WTHC5_PFC_1_quant" "WTHC5_PFC_2_quant" "WTHC5_PFC_3_quant"</span></span>
<span><span class="co">##  [4] "WTHC5_PFC_4_quant" "WTHC5_PFC_5_quant" "WTSD5_PFC_1_quant"</span></span>
<span><span class="co">##  [7] "WTSD5_PFC_2_quant" "WTSD5_PFC_3_quant" "WTSD5_PFC_4_quant"</span></span>
<span><span class="co">## [10] "WTSD5_PFC_5_quant"</span></span></code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">g_uq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HC5_1"</span>, <span class="st">"HC5_2"</span>, <span class="st">"HC5_3"</span>, <span class="st">"HC5_4"</span>, <span class="st">"HC5_5"</span>,</span>
<span>    <span class="st">"SD5_1"</span>, <span class="st">"SD5_2"</span>, <span class="st">"SD5_3"</span>, <span class="st">"SD5_4"</span>, <span class="st">"SD5_5"</span><span class="op">)</span></span></code></pre></div>
<p>Here we plot two quality control plots (RLE and PCA) following UQ
normalization. Note that triangles are sleep deprived animals and
circles are home cage control animals that were collected at the same
circadian time:</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="va">g_uq</span>, col <span class="op">=</span> <span class="va">colLib</span>, outline <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">3</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,</span>
<span>    <span class="fl">1</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Relative Log Expression"</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, cex.lab <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-64-1.png" alt="plot of chunk unnamed-chunk-64" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">g_uq</span>, labels <span class="op">=</span> <span class="cn">FALSE</span>, col <span class="op">=</span> <span class="va">colLib</span>, cex <span class="op">=</span> <span class="fl">1.5</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>,</span>
<span>    cex.lab <span class="op">=</span> <span class="fl">1.3</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>    pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">19</span>, <span class="fl">17</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-64-2.png" alt="plot of chunk unnamed-chunk-64" style="display: block; margin: auto;"></p>
<p><em>RUVseq</em>: Here we will estimate a matrix that contains
estimated factors of unwanted factors after UQ normalization.</p>
<p><em>RUVs</em> uses technical replicates or negative controls samples.
We do have negative control genes from <span class="citation">[@gerstner2016]</span> for differential gene analysis,
and will load them here. For additional information regarding
<em>RUVs</em>, see <span class="citation">[@risso2014]</span>.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"Gene_Negative_Controls.rda"</span><span class="op">)</span></span></code></pre></div>
<p>After loading negative controls, intersect with the counts, to only
include the negative controls that are expressed in the dataset:</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Negative</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">Gene_Negative_Controls</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts_gy</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">Negative</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 3024</span></span></code></pre>
<p>After viewing the results of the differential expression analysis
with several <em>k</em>s, we have set the numbers of factors of unwanted
variation (<em>k</em>) = 4, which optimizes the number of differentially
expressed genes and transcripts detected, and the positive control
recovery (at the gene level) without removing signal.</p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RUVSeq/man/RUVs.html" class="external-link">RUVs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">g_uq</span>, cIdx <span class="op">=</span> <span class="va">Negative</span>, scIdx <span class="op">=</span> <span class="va">groups</span>, k <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p>Plot the RLE and PCA plots (quality control figures) again following
<em>RUVs</em> normalization.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="va">gs</span><span class="op">$</span><span class="va">normalizedCounts</span>, col <span class="op">=</span> <span class="va">colLib</span>, outline <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="st">"Relative Log Expression"</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    cex.lab <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-68-1.png" alt="plot of chunk unnamed-chunk-68" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">gs</span><span class="op">$</span><span class="va">normalizedCounts</span>, labels <span class="op">=</span> <span class="cn">FALSE</span>, col <span class="op">=</span> <span class="va">colLib</span>, cex <span class="op">=</span> <span class="fl">1.3</span>,</span>
<span>    cex.axis <span class="op">=</span> <span class="fl">1.2</span>, cex.lab <span class="op">=</span> <span class="fl">1.2</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>, <span class="fl">0.75</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.75</span>,</span>
<span>        <span class="fl">0.75</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">19</span>, <span class="fl">17</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-68-2.png" alt="plot of chunk unnamed-chunk-68" style="display: block; margin: auto;"></p>
<p>Downstream we will use <em>gs$W</em> (from <em>RUVs</em>) as the
estimation of batch factors.</p>
<p>‘infRepIdx’ lists the inferential replicates (30):</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infRepIdx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"infRep"</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Save the number of inferential replicates as ‘nreps_gy’:</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nreps_gy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">infRepIdx</span><span class="op">)</span></span>
<span><span class="va">nreps_gy</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 30</span></span></code></pre>
<p>Account for continuous variables with <em>removeBatchEffect</em> from
<em>limma</em>. Our samples do not fall into discrete clusters so we
will use the following approach recommended by the <em>fishpond</em>
developers.</p>
<p>In short, this is done by directly scaling the estimated counts
across inferential replicates.</p>
<p>First inferential replicates are logged as <em>limma</em> requires
log-expression values for a series of samples. This is done with the
<em>assay</em> function from the <em>SummarizedExperiment</em>
package.</p>
<p>First, we load the <em>limma</em> package:</p>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_matrix_gy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">condition</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pc</span> <span class="op">&lt;-</span> <span class="fl">0.1</span>  <span class="co"># This is added to avoid negative InfReps</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">nreps_gy</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">logInfRep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">gy</span>, <span class="va">infRepIdx</span><span class="op">[</span><span class="va">k</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="va">pc</span><span class="op">)</span></span>
<span>    <span class="va">logInfRep</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/removeBatchEffect.html" class="external-link">removeBatchEffect</a></span><span class="op">(</span><span class="va">logInfRep</span>, covariates <span class="op">=</span> <span class="va">gs</span><span class="op">$</span><span class="va">W</span>,</span>
<span>        design <span class="op">=</span> <span class="va">model_matrix_gy</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">gy</span>, <span class="va">infRepIdx</span><span class="op">[</span><span class="va">k</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="va">logInfRep</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-swish-and-plot-additional-figures-1">Run Swish and Plot Additional Figures<a class="anchor" aria-label="anchor" href="#run-swish-and-plot-additional-figures-1"></a>
</h3>
<p>The Swish method is described in <span class="citation">[@zhu2019]</span>. Note that the <em>set.seed</em>
function allows for the reproducibility of exact results in the
future.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">swish</span><span class="op">(</span><span class="va">gy</span>, x <span class="op">=</span> <span class="st">"condition"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100</span></span></code></pre>
<p>View a table of differently expressed genes with a qvalue &lt;
0.05:</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">$</span><span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">##  9829  8505</span></span></code></pre>
<p>You can also view a table that shows which genes are up regulated (1)
in response to sleep deprivation and which are downregulated (-1):</p>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, sign.lfc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        sign.lfc</span></span>
<span><span class="co">## sig       -1    1</span></span>
<span><span class="co">##   FALSE 5456 4373</span></span>
<span><span class="co">##   TRUE  4364 4141</span></span></code></pre>
<p>Here, we can assess our positive control recovery:</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">significant_qvalue_gy</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">$</span><span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,</span>
<span>    <span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">significant_qvalue_gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 8505    9</span></span></code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Positive_Controls_Recovery</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">significant_qvalue_gy</span><span class="op">)</span>,</span>
<span>    <span class="va">Intersect_Positive_Expressed</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">significant_qvalue_gy</span><span class="op">)</span>, <span class="va">Intersect_Positive_Expressed</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">Intersect_Positive_Expressed</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span></span>
<span>    <span class="fl">100</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 83.15946</span></span></code></pre>
<p>Plot a histogram to show the distribution of pvalues:</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">$</span><span class="va">pvalue</span>, col <span class="op">=</span> <span class="st">"grey"</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10000</span><span class="op">)</span>, main <span class="op">=</span> <span class="st">""</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"Pvalue"</span>, cex.axis <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-76-1.png" alt="plot of chunk unnamed-chunk-76" style="display: block; margin: auto;"></p>
<p>Add symbols (Gene IDs) to the <em>SummarizedExperiment</em> object,
just like we did with transcript expression analysis.</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">addIds</span><span class="op">(</span><span class="va">gy</span>, <span class="st">"SYMBOL"</span>, gene <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 18334 rows and 10 columns</span></span>
<span><span class="co">##                                  gene_id</span></span>
<span><span class="co">##                              &lt;character&gt;</span></span>
<span><span class="co">## ENSMUSG00000000001  ENSMUSG00000000001.4</span></span>
<span><span class="co">## ENSMUSG00000000028 ENSMUSG00000000028.15</span></span>
<span><span class="co">## ENSMUSG00000000037 ENSMUSG00000000037.17</span></span>
<span><span class="co">## ENSMUSG00000000049 ENSMUSG00000000049.11</span></span>
<span><span class="co">## ENSMUSG00000000056  ENSMUSG00000000056.7</span></span>
<span><span class="co">## ...                                  ...</span></span>
<span><span class="co">## ENSMUSG00000118608  ENSMUSG00000118608.1</span></span>
<span><span class="co">## ENSMUSG00000118619  ENSMUSG00000118619.1</span></span>
<span><span class="co">## ENSMUSG00000118623  ENSMUSG00000118623.1</span></span>
<span><span class="co">## ENSMUSG00000118633  ENSMUSG00000118633.1</span></span>
<span><span class="co">## ENSMUSG00000118651  ENSMUSG00000118651.1</span></span>
<span><span class="co">##                                                                                 tx_ids</span></span>
<span><span class="co">##                                                                        &lt;CharacterList&gt;</span></span>
<span><span class="co">## ENSMUSG00000000001                                                ENSMUST00000000001.4</span></span>
<span><span class="co">## ENSMUSG00000000028 ENSMUST00000000028.13,ENSMUST00000096990.9,ENSMUST00000115585.1,...</span></span>
<span><span class="co">## ENSMUSG00000000037  ENSMUST00000238603.1,ENSMUST00000101113.8,ENSMUST00000087090.9,...</span></span>
<span><span class="co">## ENSMUSG00000000049  ENSMUST00000133383.7,ENSMUST00000146050.1,ENSMUST00000152958.7,...</span></span>
<span><span class="co">## ENSMUSG00000000056      ENSMUST00000103015.3,ENSMUST00000154047.1,ENSMUST00000151088.1</span></span>
<span><span class="co">## ...                                                                                ...</span></span>
<span><span class="co">## ENSMUSG00000118608                           ENSMUST00000232598.1,ENSMUST00000239181.1</span></span>
<span><span class="co">## ENSMUSG00000118619                                                ENSMUST00000239256.1</span></span>
<span><span class="co">## ENSMUSG00000118623                           ENSMUST00000124129.7,ENSMUST00000145993.8</span></span>
<span><span class="co">## ENSMUSG00000118633      ENSMUST00000239354.1,ENSMUST00000239230.1,ENSMUST00000239308.1</span></span>
<span><span class="co">## ENSMUSG00000118651                                                ENSMUST00000239444.1</span></span>
<span><span class="co">##                    log10mean      keep      stat       log2FC</span></span>
<span><span class="co">##                    &lt;numeric&gt; &lt;logical&gt; &lt;numeric&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">## ENSMUSG00000000001   3.06833      TRUE  12.26667    0.1757289</span></span>
<span><span class="co">## ENSMUSG00000000028   1.79075      TRUE  -0.90000   -0.0213605</span></span>
<span><span class="co">## ENSMUSG00000000037   1.63299      TRUE -12.03333   -0.8299316</span></span>
<span><span class="co">## ENSMUSG00000000049   1.19460      TRUE   2.56667    0.0461280</span></span>
<span><span class="co">## ENSMUSG00000000056   3.31778      TRUE  12.50000    0.2429778</span></span>
<span><span class="co">## ...                      ...       ...       ...          ...</span></span>
<span><span class="co">## ENSMUSG00000118608  2.043497      TRUE  -5.36667 -2.46083e-01</span></span>
<span><span class="co">## ENSMUSG00000118619  1.177520      TRUE  -5.50000 -1.62821e-01</span></span>
<span><span class="co">## ENSMUSG00000118623  1.301789      TRUE  -4.93333 -6.98220e-01</span></span>
<span><span class="co">## ENSMUSG00000118633  0.923585      TRUE  -2.40000 -3.61364e-05</span></span>
<span><span class="co">## ENSMUSG00000118651  1.396473      TRUE -10.50000 -5.64111e-01</span></span>
<span><span class="co">##                         pvalue      locfdr      qvalue</span></span>
<span><span class="co">##                      &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">## ENSMUSG00000000001 2.36337e-03 2.73538e-02 0.010738538</span></span>
<span><span class="co">## ENSMUSG00000000028 8.29876e-01 1.00000e+00 0.874171215</span></span>
<span><span class="co">## ENSMUSG00000000037 2.91153e-03 3.10390e-02 0.011396243</span></span>
<span><span class="co">## ENSMUSG00000000049 5.37582e-01 1.00000e+00 0.627612073</span></span>
<span><span class="co">## ENSMUSG00000000056 5.45435e-07 1.45352e-05 0.000003663</span></span>
<span><span class="co">## ...                        ...         ...         ...</span></span>
<span><span class="co">## ENSMUSG00000118608  0.20218501   1.0000000   0.2894401</span></span>
<span><span class="co">## ENSMUSG00000118619  0.19165703   1.0000000   0.2768547</span></span>
<span><span class="co">## ENSMUSG00000118623  0.23875805   1.0000000   0.3305437</span></span>
<span><span class="co">## ENSMUSG00000118633  0.56579633   1.0000000   0.6534780</span></span>
<span><span class="co">## ENSMUSG00000118651  0.00899858   0.0905034   0.0232366</span></span>
<span><span class="co">##                         SYMBOL</span></span>
<span><span class="co">##                    &lt;character&gt;</span></span>
<span><span class="co">## ENSMUSG00000000001       Gnai3</span></span>
<span><span class="co">## ENSMUSG00000000028       Cdc45</span></span>
<span><span class="co">## ENSMUSG00000000037       Scml2</span></span>
<span><span class="co">## ENSMUSG00000000049        Apoh</span></span>
<span><span class="co">## ENSMUSG00000000056        Narf</span></span>
<span><span class="co">## ...                        ...</span></span>
<span><span class="co">## ENSMUSG00000118608      Gm4988</span></span>
<span><span class="co">## ENSMUSG00000118619          NA</span></span>
<span><span class="co">## ENSMUSG00000118623          NA</span></span>
<span><span class="co">## ENSMUSG00000118633   Ccl19-ps2</span></span>
<span><span class="co">## ENSMUSG00000118651     Gm17315</span></span></code></pre>
<p>Turn ‘gy’ into a dataframe here, and save as a different variable. We
do this to intersect with other lists when making plots, such as genes
to highlight on the MA plot downstream. Check the <em>dim</em> to ensure
that it has not changed:</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gy_rowData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gy_rowData</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 18334    10</span></span></code></pre>
<p>Here we graph an MA plot. An MA plot shows log2FC vs the log10mean.
Genes above M=0 are upregulated, while genes below are
downregulated.</p>
<p>The genes that are farthest away from the y intercept (have the
largest log10mean values) are the ones most affected by the
treatment.</p>
<p>We previously set colors for the MA plot, so we will only set one
additional color here: the positive controls.</p>
<p>A quick summary of the other colors: Genes that do not have a
significant change in expression are light grey. Genes that are
significantly affected (qvalue &lt; 0.05) by SD are dark grey. Genes
that we have chosen to highlight that are also significant are bright
blue.</p>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Positive_Color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<p>We will intersect ‘gy_rowData’ (the dataframe saved above) with the
‘Gene_Names’ matrix from transcript expression analysis:</p>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Data_Intersection</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gy_rowData</span>, <span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">Gene_Names</span> <span class="op">&amp;</span></span>
<span>    <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 10 10</span></span></code></pre>
<p>Use <em>plotMASwish</em> to generate the MA plot. Include labels and
points for the genes in ‘Gene_Names’ and points for positive controls
using the code below (Only genes that fit the qvalue and log2FC
threshold will be labeled):</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotMASwish</span><span class="op">(</span><span class="va">gy</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">5.5</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">7</span>,</span>
<span>    <span class="fl">5</span><span class="op">)</span>, sigcolor <span class="op">=</span> <span class="va">Significant_Color</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, cex.lab <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span>, <span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">Gene_Names</span> <span class="op">&amp;</span> <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">log2FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="va">log10mean</span>, <span class="va">log2FC</span>, <span class="va">SYMBOL</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>,</span>
<span>    pos <span class="op">=</span> <span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">0.8</span>, font <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Positive_Controls_Signifcant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">significant_qvalue_gy</span><span class="op">)</span>,</span>
<span>    <span class="va">Intersect_Positive_Expressed</span><span class="op">)</span></span>
<span><span class="va">Positive_Controls_Plot</span> <span class="op">&lt;-</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span></span>
<span>    <span class="va">Positive_Controls_Signifcant</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Positive_Controls_Plot</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 558  10</span></span></code></pre>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Positive_Controls_Plot</span><span class="op">$</span><span class="va">log10mean</span>, y <span class="op">=</span> <span class="va">Positive_Controls_Plot</span><span class="op">$</span><span class="va">log2FC</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">20</span>, col <span class="op">=</span> <span class="va">Positive_Color</span>, cex <span class="op">=</span> <span class="fl">0.2</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log10mean</span>, <span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log2FC</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>, cex <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log10mean</span>, <span class="va">Data_Intersection</span><span class="op">$</span><span class="va">log2FC</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">20</span>, col <span class="op">=</span> <span class="va">Highlight_Color</span>, cex <span class="op">=</span> <span class="fl">0.2</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="figure/unnamed-chunk-81-1.png" alt="plot of chunk unnamed-chunk-81" style="display: block; margin: auto;"></p>
<p>Write text file using the code below. Select columns to include in
the text file. These lists will then be annotated in Perl with the gene
name and description.</p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"log10mean"</span>, <span class="st">"log2FC"</span>, <span class="st">"pvalue"</span>, <span class="st">"qvalue"</span><span class="op">)</span></span>
<span><span class="va">gy_filter</span> <span class="op">&lt;-</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span>
<span><span class="va">gy_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gy_filter</span><span class="op">)</span><span class="op">[</span>, <span class="va">cols</span><span class="op">]</span></span>
<span><span class="va">gy_filter</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gy_filter</span>, <span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gy_filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 8505    4</span></span></code></pre>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">gy_filter</span>, file <span class="op">=</span> <span class="st">"DGE_Significant_k=4.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Ventura 13.2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [7] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] GenomicFeatures_1.50.4      org.Mm.eg.db_3.16.0        </span></span>
<span><span class="co">##  [3] AnnotationDbi_1.60.2        RUVSeq_1.32.0              </span></span>
<span><span class="co">##  [5] edgeR_3.40.2                limma_3.54.2               </span></span>
<span><span class="co">##  [7] EDASeq_2.32.0               ShortRead_1.56.1           </span></span>
<span><span class="co">##  [9] GenomicAlignments_1.34.1    Rsamtools_2.14.0           </span></span>
<span><span class="co">## [11] Biostrings_2.66.0           XVector_0.38.0             </span></span>
<span><span class="co">## [13] BiocParallel_1.32.6         RColorBrewer_1.1-3         </span></span>
<span><span class="co">## [15] fishpond_2.4.1              tximeta_1.16.1             </span></span>
<span><span class="co">## [17] SummarizedExperiment_1.28.0 Biobase_2.58.0             </span></span>
<span><span class="co">## [19] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        </span></span>
<span><span class="co">## [21] IRanges_2.32.0              S4Vectors_0.36.2           </span></span>
<span><span class="co">## [23] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      </span></span>
<span><span class="co">## [25] matrixStats_0.63.0          knitr_1.42                 </span></span>
<span><span class="co">## [27] BiocStyle_2.26.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] readxl_1.4.2                  AnnotationHub_3.6.0          </span></span>
<span><span class="co">##   [3] aroma.light_3.28.0            BiocFileCache_2.6.1          </span></span>
<span><span class="co">##   [5] shades_1.4.0                  plyr_1.8.8                   </span></span>
<span><span class="co">##   [7] lazyeval_0.2.2                splines_4.2.2                </span></span>
<span><span class="co">##   [9] ggplot2_3.4.2                 digest_0.6.31                </span></span>
<span><span class="co">##  [11] ensembldb_2.22.0              htmltools_0.5.5              </span></span>
<span><span class="co">##  [13] fansi_1.0.4                   magrittr_2.0.3               </span></span>
<span><span class="co">##  [15] memoise_2.0.1                 svMisc_1.2.3                 </span></span>
<span><span class="co">##  [17] R.utils_2.12.2                prettyunits_1.1.1            </span></span>
<span><span class="co">##  [19] jpeg_0.1-10                   colorspace_2.1-0             </span></span>
<span><span class="co">##  [21] blob_1.2.4                    rappdirs_0.3.3               </span></span>
<span><span class="co">##  [23] xfun_0.38                     dplyr_1.1.1                  </span></span>
<span><span class="co">##  [25] crayon_1.5.2                  RCurl_1.98-1.12              </span></span>
<span><span class="co">##  [27] jsonlite_1.8.4                tximport_1.26.1              </span></span>
<span><span class="co">##  [29] glue_1.6.2                    gtable_0.3.3                 </span></span>
<span><span class="co">##  [31] zlibbioc_1.44.0               DelayedArray_0.24.0          </span></span>
<span><span class="co">##  [33] SingleCellExperiment_1.20.1   abind_1.4-5                  </span></span>
<span><span class="co">##  [35] scales_1.2.1                  DBI_1.1.3                    </span></span>
<span><span class="co">##  [37] Rcpp_1.0.10                   xtable_1.8-4                 </span></span>
<span><span class="co">##  [39] progress_1.2.2                bit_4.0.5                    </span></span>
<span><span class="co">##  [41] httr_1.4.5                    ellipsis_0.3.2               </span></span>
<span><span class="co">##  [43] pkgconfig_2.0.3               XML_3.99-0.14                </span></span>
<span><span class="co">##  [45] R.methodsS3_1.8.2             dbplyr_2.3.2                 </span></span>
<span><span class="co">##  [47] deldir_1.0-6                  locfit_1.5-9.7               </span></span>
<span><span class="co">##  [49] utf8_1.2.3                    tidyselect_1.2.0             </span></span>
<span><span class="co">##  [51] rlang_1.1.0                   reshape2_1.4.4               </span></span>
<span><span class="co">##  [53] later_1.3.0                   munsell_0.5.0                </span></span>
<span><span class="co">##  [55] BiocVersion_3.16.0            cellranger_1.1.0             </span></span>
<span><span class="co">##  [57] tools_4.2.2                   cachem_1.0.7                 </span></span>
<span><span class="co">##  [59] cli_3.6.1                     generics_0.1.3               </span></span>
<span><span class="co">##  [61] RSQLite_2.3.1                 evaluate_0.20                </span></span>
<span><span class="co">##  [63] stringr_1.5.0                 fastmap_1.1.1                </span></span>
<span><span class="co">##  [65] yaml_2.3.7                    bit64_4.0.5                  </span></span>
<span><span class="co">##  [67] purrr_1.0.1                   KEGGREST_1.38.0              </span></span>
<span><span class="co">##  [69] AnnotationFilter_1.22.0       mime_0.12                    </span></span>
<span><span class="co">##  [71] formatR_1.14                  R.oo_1.25.0                  </span></span>
<span><span class="co">##  [73] xml2_1.3.3                    biomaRt_2.54.1               </span></span>
<span><span class="co">##  [75] compiler_4.2.2                rstudioapi_0.15.0            </span></span>
<span><span class="co">##  [77] filelock_1.0.2                curl_5.0.0                   </span></span>
<span><span class="co">##  [79] png_0.1-8                     interactiveDisplayBase_1.36.0</span></span>
<span><span class="co">##  [81] tibble_3.2.1                  stringi_1.7.12               </span></span>
<span><span class="co">##  [83] highr_0.10                    lattice_0.21-8               </span></span>
<span><span class="co">##  [85] ProtGenerics_1.30.0           Matrix_1.5-4                 </span></span>
<span><span class="co">##  [87] vctrs_0.6.1                   pillar_1.9.0                 </span></span>
<span><span class="co">##  [89] lifecycle_1.0.3               BiocManager_1.30.20          </span></span>
<span><span class="co">##  [91] bitops_1.0-7                  httpuv_1.6.9                 </span></span>
<span><span class="co">##  [93] patchwork_1.1.2               rtracklayer_1.58.0           </span></span>
<span><span class="co">##  [95] qvalue_2.30.0                 R6_2.5.1                     </span></span>
<span><span class="co">##  [97] BiocIO_1.8.0                  latticeExtra_0.6-30          </span></span>
<span><span class="co">##  [99] hwriter_1.3.2.1               promises_1.2.0.1             </span></span>
<span><span class="co">## [101] codetools_0.2-19              MASS_7.3-58.3                </span></span>
<span><span class="co">## [103] gtools_3.9.4                  rjson_0.2.21                 </span></span>
<span><span class="co">## [105] withr_2.5.0                   GenomeInfoDbData_1.2.9       </span></span>
<span><span class="co">## [107] parallel_4.2.2                hms_1.1.3                    </span></span>
<span><span class="co">## [109] grid_4.2.2                    rmarkdown_2.21               </span></span>
<span><span class="co">## [111] shiny_1.7.4                   interp_1.1-4                 </span></span>
<span><span class="co">## [113] restfulr_0.0.15</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Elena Zuin, Kaitlyn Ford, Dario Righelli, Davide Risso.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
